src/network/network.cpp

```diff
@@ -2367,6 +2403,41 @@ const utf8 * network_get_server_provider_name() { return gNetwork.ServerProvider
 const utf8 * network_get_server_provider_email() { return gNetwork.ServerProviderEmail.c_str(); }
 const utf8 * network_get_server_provider_website() { return gNetwork.ServerProviderWebsite.c_str(); }
 
+static bool network_save_cheatsDisableClearanceChecks;
+
+void network_save_server_prefs() {
+	if (network_get_mode() != NETWORK_MODE_SERVER) {
+		return;
+	}
+	network_save_cheatsDisableClearanceChecks = gCheatsDisableClearanceChecks;
+}
+
+void network_apply_client_prefs(int game_command_playerid) {
+	NetworkPlayer *player = gNetwork.GetPlayerByID(game_command_playerid);
+	if (player == nullptr) {
+		return;
+	}
+	for (const auto &it : player->user_prefs) {
+		switch (it.first) {
+			case CLIENT_PREFS::CLIENT_PREF_DISABLE_CLEARENCE:
+				// Even if a server does not allow using disable clearance and so this cannot be *set*,
+				// a client can still have this setting on, so we need to check anyway.
+				if (gServerCheatsDisableClearanceChecks) {
+					gCheatsDisableClearanceChecks = !!(it.second);
```

Comment:
`!!`, this like the sort cryptic syntax you see in javascript. Why not `it.second != 0`?